Updated documentation of Auth module (#683)
EVerest/everest-core: modules/Auth/docs/index.rst

.. _everest_modules_handwritten_Auth:

===========
Auth Module
===========

This module handles incoming authorization and reservation requests.

The task of the module is to receive tokens from token providers, validate them and assign them to EvseManagers.

> It is responsible to provide authorization to EvseManagers and to stop transactions at the EvseManagers if a token

Reviewer: -> responsible for providing ... and for stopping

or parent id token is presented to stop a transaction. In addition, the module is responsible for managing all
reservations and matching them with incoming tokens.

The module contains the logic to select a connector for incoming tokens (e.g. by waiting for a car plug in, user
interface, random selection, etc.). Currently two selection algorithms are implemented and described in 
`Selection Algorithm`_.

The following flow diagram describes how an incoming token is handled by the module.

.. image:: token_handling.drawio.svg
   :alt: TokenHandling

.. note::
    
    The processing of each authorization request and the respective validation runs in an individual thread. This 
    allows the parallel processing of authorization requests.

Integration in EVerest
======================

> This module provides implementation for the reservation and the auth interface.
> 
> It requires connections to module(s) implementing the token_provider, token_validator and evse_manager interfaces.

Reviewer: Perhaps mark the interface names with markup, just to make it clear they are not plain text?

Also implementation -> implementations

```suggestion
This module provides implementations for the `reservation` and the `auth` interfaces.

It requires connections to module(s) implementing the `token_provider`, `token_validator` and `evse_manager` interfaces.
```

The following diagram shows how it integrates with other EVerest modules.

.. image:: everest_integration.drawio.svg
   :alt: Integration

Selection Algorithm
===================

The selection algorithm contains the logic to select one connector for an incoming token. The algorithm can be
specified within the module config using the key `selection_algorithm`. In case the charging station has only 
one connector, the selection of a connector is pretty straight-forward because there is only one that is 

> available. The selection algorithm becomes relevant in case the Auth  module manages authorization requests 

Reviewer: Stray space.
```suggestion
available. The selection algorithm becomes relevant in case the Auth module manages authorization requests 
```

for multiple connectors. 

Three options for the selection are available: 

* PlugEvents
* FindFirst
* UserInput

PlugEvents
----------

The following flow chart describes how a connector is selected using the `PlugEvents` algorithm.

.. image:: plug_events_selection_algorithm.drawio.svg
   :alt: SelectionAlgorithm

.. note::
    
    In case a user authorizes first and no EV is connected to the charger to initiate a SessionStarted event the 

>     processing thread waits for a SessionStarted event to select the respective connector. A Plug-In  timeout may 

Reviewer: Stray space
```suggestion
    processing thread waits for a SessionStarted event to select the respective connector. A Plug-In timeout may 
```
